transactor/operator.go

```diff
@@ -0,0 +1,148 @@
+package transactor
+
+import (
+	"math/big"
+
+	"github.com/celer-network/goutils/log"
+	"github.com/celer-network/sgn/common"
+	"github.com/celer-network/sgn/mainchain"
+	"github.com/celer-network/sgn/x/sync"
+	"github.com/celer-network/sgn/x/validator"
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/staking"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/spf13/viper"
+)
+
+type Operator struct {
+	EthClient  *mainchain.EthClient
+	Transactor *Transactor
+}
+
+func NewOperator(cdc *codec.Codec) (operator Operator, err error) {
+	ethClient, err := mainchain.NewEthClient(
+		viper.GetString(common.FlagEthGateway),
+		viper.GetString(common.FlagEthKeystore),
+		viper.GetString(common.FlagEthPassphrase),
+		&mainchain.TransactorConfig{
+			BlockDelay:           viper.GetUint64(common.FlagEthBlockDelay),
+			BlockPollingInterval: viper.GetUint64(common.FlagEthPollInterval),
+			ChainId:              big.NewInt(viper.GetInt64(common.FlagEthChainID)),
+			AddGasPriceGwei:      viper.GetUint64(common.FlagEthAddGasPriceGwei),
+			MinGasPriceGwei:      viper.GetUint64(common.FlagEthMinGasPriceGwei),
+		},
+		viper.GetString(common.FlagEthDPoSAddress),
+		viper.GetString(common.FlagEthSGNAddress),
+		viper.GetString(common.FlagEthLedgerAddress),
+	)
+	if err != nil {
+		return
+	}
+
+	transactor, err := NewTransactor(
+		viper.GetString(common.FlagCLIHome),
+		viper.GetString(common.FlagSgnChainID),
+		viper.GetString(common.FlagSgnNodeURI),
+		viper.GetString(common.FlagSgnOperator),
+		viper.GetString(common.FlagSgnPassphrase),
+		cdc,
+		NewGasPriceEstimator(viper.GetString(common.FlagSgnNodeURI)),
+	)
+	if err != nil {
+		return
+	}
+
+	return Operator{
+		EthClient:  ethClient,
+		Transactor: transactor,
+	}, nil
+}
+
+func (o Operator) SyncUpdateSidechainAddr(candidateAddr mainchain.Addr) {
```

Comment:
`(o *Operator)`, same for all other places

Reply:
why use o *Operator?

Reply:
more efficient and idiomatic